lib/vquic/ngtcp2.c

```diff
@@ -1666,21 +1666,23 @@ static CURLcode do_sendmsg(size_t *psent, struct Curl_easy *data, int sockfd,
   msg.msg_iov = &msg_iov;
   msg.msg_iovlen = 1;
 
-  msg.msg_control = msg_ctrl;
-  msg.msg_controllen = sizeof(msg_ctrl);
-
 #if defined(__linux__) && defined(UDP_SEGMENT)
   if(pktlen > gsolen) {
+    /* Only set this, when we need it. macOS, for example,
+     * does not seem to like a msg_control of length 0. */
+    msg.msg_control = msg_ctrl;
+    msg.msg_controllen = sizeof(msg_ctrl);
+
     ctrllen += CMSG_SPACE(sizeof(uint16_t));
```

Comment:
I'm not familiar with how these macros work but I wonder is separate `ctrllen` still needed now, for example would this be correct `msg.msg_controllen = sizeof(msg_ctrl) + CMSG_SPACE(sizeof(uint16_t));` ?

Reply:
hm if not, how about this:
msg.msg_controllen = sizeof(msg_ctrl);
msg.msg_controllen += CMSG_SPACE(sizeof(uint16_t));

Reply:
Because we use UDP_SEGMENT only, just assigning `CMSG_SPACE(sizeof(uint16_t))` to `msg.msg_controllen` should work:

```c
    msg.msg_control = msg_ctrl;
    msg.msg_controllen = CMSG_SPACE(sizeof(uint16_t));
```

Reply:
Just added that change and cleaned up the local var and assertion.